packages/dev/serializers/src/glTF/2.0/glTFExporter.ts

```diff
@@ -422,6 +449,288 @@ export class _Exporter {
+                                (vertexData as Vector4[]).push(Vector4.FromArray(meshAttributeArray, index + stride));
+                            } else {
+                                (vertexData as Vector3[]).push(Vector3.FromArray(meshAttributeArray, index));
+                                (vertexData as Vector3[]).push(Vector3.FromArray(meshAttributeArray, index + 2 * stride));
+                                (vertexData as Vector3[]).push(Vector3.FromArray(meshAttributeArray, index + stride));
+                            }
+                        }
+                        break;
+                    }
+                    case VertexBuffer.UVKind:
+                    case VertexBuffer.UV2Kind: {
+                        for (let x = submesh.verticesStart; x < submesh.verticesStart + submesh.verticesCount; x = x + 3) {
+                            index = x * stride;
+                            (vertexData as Vector2[]).push(Vector2.FromArray(meshAttributeArray, index));
+                            (vertexData as Vector2[]).push(Vector2.FromArray(meshAttributeArray, index + 2 * stride));
+                            (vertexData as Vector2[]).push(Vector2.FromArray(meshAttributeArray, index + stride));
+                        }
+                        break;
+                    }
+                    default: {
+                        Tools.Error(`Unsupported Vertex Buffer type: ${vertexBufferKind}`);
+                    }
+                }
+                this._writeVertexAttributeData(vertexData, byteOffset, vertexBufferKind, binaryWriter);
+            }
+        } else {
+            Tools.Warn(`reorderTriangleFillMode: Vertex Buffer Kind ${vertexBufferKind} not present!`);
+        }
+    }
+
+    /**
+     * Reorders the vertex attributes in the correct triangle strip order.  This is necessary when indices are not available and the winding order is
+     * clock-wise during export to glTF
+     * @param submesh BabylonJS submesh
+     * @param vertexBufferKind The type of vertex attribute
+     * @param meshAttributeArray The vertex attribute data
+     * @param byteOffset The offset to the binary data
+     * @param binaryWriter The binary data for the glTF file
+     */
+    private _reorderTriangleStripDrawMode(submesh: SubMesh, vertexBufferKind: string, meshAttributeArray: FloatArray, byteOffset: number, binaryWriter: _BinaryWriter) {
+        const vertexBuffer = this._getVertexBufferFromMesh(vertexBufferKind, submesh.getMesh() as Mesh);
+        if (vertexBuffer) {
+            const stride = vertexBuffer.byteStride / VertexBuffer.GetTypeByteLength(vertexBuffer.type);
+
+            const vertexData: Vector2[] | Vector3[] | Vector4[] = [];
+            let index = 0;
+            switch (vertexBufferKind) {
+                case VertexBuffer.PositionKind:
+                case VertexBuffer.NormalKind: {
+                    index = submesh.verticesStart;

```

Comment:
Shouldn't you perform the same operations for position/normal as for tangent/color/uv/uv2? Data will no longer be aligned if position/normal exchanges only the first and second values, whereas tangent/color/uv/uv2 rewrites all values in reverse order.

Reply:
Yes, I thought about this too when looking at it. It definitely seems like values will be mismatched afterwards, but it's existing code like I mentioned for the other comment.